### DISCUSSION TITLE
What is the best desktop client?

### DISCUSSION THREAD

**Discussion by digitaldissent** [TechbeeAT/jtxBoard#542] [Q&A]
I'd like to know what the recommended (Linux)desktop client is; none that I have tried work well for me. I tried using Nextcloud as a backend and then synched to Kmail and Thunderbird. Tasks would "pile up" in my email client and not sync properly. I could add a task from my email client, and it would sync to Jtx; then, if I checked it done in Jtx, it would never be marked done in Thunderbird/Kmail. Tasks just seem to get stuck and then pile up. If I created and then complete a task in Jtx it might sync all that to Thunderbird or get stuck as an open task or maybe get stuck as a completed task.

I moved to Radicale as a backend, and now with Thunderbird and Evolution, the same issues. Evolution works a little better.

I am not getting any errors in my clients, and I am on Jtx 2.4

What are people using as desktop clients? Jtx alone is fine, but I want to use my laptop : /

**Comment by rogercreagh** [comment]
Good question. Currently I haven't found one - for me the problem is that I primarily use journals - tasks are well handled by a plethora of apps - one of jtxB's main benefits for tasks is to sync through a CalDAV server, so its disappointing to learn that you have problems syncing with Tbird and Evolution.

I have used both as email & calendar clients and in themselves I found their tasks apps ok for laptop use but never managed to get them syncing well with a phone app. But basically I'm too disorganised to use a task app on any platform effectively anyway.

For journals I don't have a desktop solution at all - there is (as yet) no linux app that works with a CalDAV server for Vcalendar journal entries without using a calendar interface. (probably the same on Win & Mac)

What is needed is a dedicated Journal and Notebook program that is optimised for handling Journals (which typically you want to read like a book or diary and add entries at the end). There are plenty of notebook apps for Linux, but none  I have found using VCalendar Vjournal entries (with no start date - which Patrick is cleverly using to distinguish between a journal/diary entry and a freestanding note in the absence of a VNOTE component in the spec) as a cross platform sync mechanism. jtxB is excellent for this  - we just need a Linux app.

My C is very rusty and I've never written a program on Linux (I started on MsDos).  I used to do a lot of php before retiring a few years ago so I am playing with building a web based Journal and Notebook viewer using php. Slow progress but nearly usable and will at least enable  me to read, edit, create, and sync entries from my laptop (when online) (or any browser on any platform).

I'd love to hear what others are using,especially those who are using jtxB to keep a journal or diary, or as a note taking device.

**Comment by patrickunterwegs** [comment]
Hey @digitaldissent ,
I have tried the synchronization with Thunderbird (on windows) and kOrganizer (Linux) and Evolution (Linux) and I didn't notice any sync issues.

As @rogercreagh already mentioned, I'm also not aware of any installable alternatives. Theoretically you could install jtx Board on Linux (there is a "show and tell" discussion about this) using Waydroid.

The backend itself shouldn't really make a difference... Radicale or Nextcloud, both should work properly. (In Nextcloud you would also have a web interface for tasks, but the functionality is very limited). 

Are you sure that your problem isn't just a sync setting issue? When looking at your description it looks more like the downwards sync from your server to thunderbird doesn't work properly. Or did you track it down to the behavior that only the completed status wasn't synchronized?

But are you sure that there wasn't just

**Reply by digitaldissent** [reply]
_When looking at your description it looks more like the downwards sync from your server to thunderbird doesn't work properly_

Indeed it does, none of these issues seem to be coming from JTX but different types of tasks appear in the clients, but the tasks are not the same, some are one-time and some are reoccurring.

I think I will setup Radicale locally and see how that works

But 1st I found this -> https://github.com/flozz/calcleaner

**Reply by digitaldissent** [reply]
@patrickunterwegs 
This is interesting, logs from my Radicale server

`Bad PUT request on '/user/8c697e24-b34c-25c4-2426-9dc2b0792806/d8aa7fbd-0333-4f0b-96d8-a6fca6247c7f.ics': Main component missing

[2023-05-08 13:28:07 +0000] [6/Thread-1910 (process_request_thread)] [WARNING] Bad PUT request on '/user/8c697e24-b34c-25c4-2426-9dc2b0792806/d8aa7fbd-0333-4f0b-96d8-a6fca6247c7f.ics': Main component missing

[2023-05-08 13:29:17 +0000] [6/Thread-1917 (process_request_thread)] [WARNING] Bad PUT request on '/user/8c697e24-b34c-25c4-2426-9dc2b0792806/d8aa7fbd-0333-4f0b-96d8-a6fca6247c7f.ics': can't compare offset-naive and offset-aware datetimes

[2023-05-08 13:32:00 +0000] [6/Thread-2017 (process_request_thread)] [WARNING] Bad PUT request on '/user/8c697e24-b34c-25c4-2426-9dc2b0792806/d8aa7fbd-0333-4f0b-96d8-a6fca6247c7f.ics': can't compare offset-naive and offset-aware datetimes

[2023-05-08 13:32:12 +0000] [6/Thread-2018 (process_request_thread)] [WARNING] Bad PUT request on '/user/8c697e24-b34c-25c4-2426-9dc2b0792806/d8aa7fbd-0333-4f0b-96d8-a6fca6247c7f.ics': Main component missing`

**Reply by patrickunterwegs** [reply]
Hm, yes, that's interesting, maybe it's really just a broken entry that messes with the whole synchronization. Would you mind uploading this problematic .ics? (I mean only if it doesn't contain any sensitive data)

**Reply by digitaldissent** [reply]
> Hm, yes, that's interesting, maybe it's really just a broken entry that messes with the whole synchronization. Would you mind uploading this problematic .ics? (I mean only if it doesn't contain any sensitive data)

A little background information, I have been hosting my Nextcloud server for a few years and then migrated my Calendar to Radicale, so there are over 2,000 .ics files many are probably corrupted. I ran the CalDav cleaner I linked to above to remove all but the last two weeks. I did notice that clients would hold onto bad entries, I deleted those and so far things are working pretty well, I am going to monitor and see how things go.

**Reply by patrickunterwegs** [reply]
Thanks for the update! Fingers crossed that the cleanup solved the issue for good 🤞😉

**Comment by bepolymathe** [comment]
Hi @all 

After various trials, I've found thunderbird to be the most convincing, because it displays tasks in the calendar... which also lets you "timeblock". However, I can't find a solution (either native or via add-ons) to retrieve my jtx memos/remarks in thunderbird (or even under linux in general). Any ideas?

PS: Evolution does it, but it's a calamity for mail retrieval compared to thunderbird...
PS2: Perhaps the jtx team could usefully develop a functional addon for Thunderbird to bring this great tool to this ecosystem (and vice versa) ;-)

**Reply by patrickunterwegs** [reply]
Hey, kOrganizer is the second Linux app that supports vjournal, but it's also very basic. Regrettably I can't recommend anything else and I'm not aware of any other solutions at the moment, but I hope others will follow.

If there was a team that's not just me, I'd definitely consider trying an add-on for another application 😉

**Reply by rogercreagh** [reply]
You are already a star for your seemingly tireless work on this. Thank you so much. I appreciate how much work is involved making slow progress on my own vjournal joomla extension. Blooming well done.

If anyone on here has any Thunderbird addon or Linux desktop app dev knowledge we would be happy to help with advice and experience from other platforms using CalDAV clients. Even drafting a requirements spec would be a start...

**Reply by bepolymathe** [reply]
Hi @patrickunterwegs and @rogercreagh

> If there was a team that's not just me, I'd definitely consider trying an add-on for another application wink

Yes, of course I understand completely. I agree that the work done is already brilliant and much appreciated. Personally, I'd be very happy with a Thunderbird addon, but I think some users would prefer a dedicated desktop application (flatpak). In terms of note-taking software, I know that Qownnotes lets you sync caldav tasks, but that's pretty basic too. 

> If anyone on here has any Thunderbird addon or Linux desktop app dev knowledge we would be happy to help with advice and experience from other platforms using CalDAV clients.

Unfortunately I have no development skills in this area. From my point of view, the advantage of a desktop application would be to be able to retrieve notes in markdown format in a folder. This would make it possible to use these notes in documentation management applications such as Logseq, Obsidian, Qownnotes, etc. But I'll have to take a look at Vdirsyncer, which was mentioned above, to see if we can achieve anything with it concerning notes or even journal entries.

**Comment by boxallar** [comment]
Hi, I have managed to sync thunderbird too. It works for tasks, but notes are not there.
Evolution is similar for me. I can sync tasks back and forth without problems, but the "Memos" tab does not seem to be syncable.
Korganizer is promising.  There are tabs/menu items for ToDos and Journals. I have not been able to sync though. Thunderbird and Evolution both have nice wizards that walked me through the setup of my account. Korganizer has some help, but I have not managed to get it to work.
Can anyone help me? I am syncing with DAVx5

**Reply by boxallar** [reply]
I found the solution to my syncing with Korganiser issues here: https://jtx.techbee.at/troubles-with-the-synchronization-see-what-could-go-wrong

RTFM, yeah, I know 🫢

I needed to create a new calendar with all the necessary entry types. It was very easy to do with DAVx⁵. A nice advantage of working closely together I guess.
In lieu of a wizard fetching the info I needed, I could copy the appropriate URLs that Thunderbird and Evolution had found.

**Reply by patrickunterwegs** [reply]
Hey, sorry for the late reply, but you figured it out anyway :-) I hope now it works smoothly :-)

**Comment by Savanta** [comment]
Thunderbird is good and pretty fast with syncing but regrettably doesn't support subtasks. Everything is piled together.
Feature request for thunderbird subtasks is 21yo... It could legally go to bar

**Comment by boxallar** [comment]
Hi, can people who use jtx on their phones and want access on desktop please share their workflows? I am still looking for a good solution. I am on Linux.

- Thunderbird has no notes or journals and the tasks do not sync with hierarchy
- Evolution seems to work for notes, but the task hierarchy and journals are not there
- Korganizer has task hierarchy and journals, but no notes as far as I can see
- I can see my tasks with hierarchy from my internet browser, but have not figured out how to see my notes there
- I am not keen on emulating a phone on my computer. It sounds inefficient, fiddly and not the best way to display things on a larger screen. I am open to being told otherwise though.

The most important things for me are notes and tasks with hierarchy of sub-tasks. Without the sub-tasks my to-dos get too long, and I have no overview. Ideally I would have everything in one application, but 2 small apps would be fine. I do not really want to use 2 big programmes like say Korganizer and Evolution.  One or two tabs in my browser would also be fine.
Any thought or suggestions will be appreciated. :)

**Reply by Julianoe** [reply]
Did you need to do something specific to make it work with Korganizer? Mine shows up empty of journals entries even though I have plenty in my Nextcloud which is synchronized with DAV

**Comment by Julianoe** [comment]
It's been a few months, any new ideas? How do you edit your journals on desktop?

**Comment by mcastillof** [comment]
You may be interested in #1554

**Reply by JurgenCruz** [reply]
I remember there was a wayland emulator for android, or something like that, that could run android apps in linux. not ideal though.

**Comment by edwin-jones** [comment]
Hey folks, if anybody needs a work around gnome evolution _technically_ supports all things, including journals. It treats notes and journals as the same and from what I can tell VJOURNAL is used for both - the difference being journal entries (that show up as such in jtx) have the date set. To convert a note to a journal entry just add the date. To convert a journal entry to a note, remove the date. I am not sure how to convert to and from tasks though.

I hope this helps folks, I haven't found a better desktop linux tool yet myself for CALDAV sync of contacts, notes, calendars, tasks and journal entries yet. No decent web solutions either.

**Reply by mcastillof** [reply]
@edwin-jones I'm trying Evolution. If a task is created in jtxBoard or Evolution, is it shown in the other application as well? Doesn't seem to work for me.

**Reply by mcastillof** [reply]
Sorry. Tasks are working fine, I confused an appointment in the calendar as a task. I would love to see tasks in the calendar. that's not posible, right?

**Reply by edwin-jones** [reply]
Not that's I'm aware of but I don't use them. I can create tasks in evolution and see them in jtx and vice versa though.

**Reply by patrickunterwegs** [reply]
The problem with showing a task in a calendar is the structure of the VEVENT and VTODO, while the VEVENT MUST have a DTSTART and a DTEND date/time (where systems can rely that they are always present), the VTODO CAN have a DTSTART, DUE and COMPLETED date/time. 
So calendar apps would struggle if there's no date at all present for a VTODO. Then a decision must be made when tasks actually end in a calendar (do they have a default duration? - which is not defined, do they end on the due date? the completed date?). There standard doesn't define this, so even if a solution is possible, I assume that most calendar apps didn't want to take such a custom decision and rather not show VTODOs.

**Reply by JurgenCruz** [reply]
The problem with Evolution is it doesn't have nested tasks and the bug report to add them has been open for 20 years. :cry:

**Comment by rogercreagh** [comment]
I haven't tried it (yet) but if you are on linux then I believe KDE Organizer which runs on the KDE Plasma desktop does handle VJOURNAL entries.

I'm on Ubuntu 24.04 and it says you can install KDE plasma desktop in parallel with Gnome (the default) and switch between them. I'm thinking I might give it a go...

**Reply by Julianoe** [reply]
Yes I've been using kOrganizer and it works well. The user interface takes a little bit to get used to, but once you understand how to switch from calendars to journals and displays everything in a weekly or monthly manner it is super useful.
In fact most my entries I do on kOrganizer now. The typing in jtxBoard is super annoying and it is in fact a quite bad experience to add entries in it as a journaling too.
In principle you should be able to install kOrganizer on Ubuntu. I'm on Manjaro and it works very well

**Reply by rogercreagh** [reply]
Thanks - I'll def give it a go.

**Reply by edwin-jones** [reply]
kOrganiser works but it doesn't handle all the other things. If all you need is journals it's probably the best desktop option but I'm on Ubuntu so I prefer native gnome apps.

**Reply by rogercreagh** [reply]
Yep, I can confirm that Korganiser/Kontact can be installed and work on Ubuntu 24.04.1 with the standard gnome desktop, so no need to switch to full KDE (although there may be advantages to doing that)

The showstopper for me is that it doesn't seem to use or display or edit Categories (tags) which are essential for organising a large collection of journal entries.

**Comment by rogercreagh** [comment]
Meanwhile one day I'll get back to my half completed Journal extension for joomla which will give ma a browser tab view of Journals and notes (but not tasks - there are other apps for them and for me tasks are a completely different application than journals - tasks are about what you are planning to do, journals are about recording what you' did.

For me I'd prefer JtxB didn't waste its time with tasks, and concentrated on its original purpose - handling VJOURNAL events well.

**Comment by ergodicbreak** [comment]
Hello everyone, I just started using CalDav as I'm trying getting my personal accounts out of Google. In searching for CalDav clients I found jtxBoard, it looked very interesting, and so of course in following the white rabbit I found this thread. I'd like to add where I'm at and see if anyone has recently come up with a good workflow for vjournal specifically that includes mobile and desktop. Tasks and events seem to be handled well by other apps. 

Unfortunately my new mail provider (Mailbox.org) doesn't seem to support vjournal with their CalDav server. So I've set up Radicale just on my laptop and am syncing that with Davx5, and using vdirsyncer to pull the .ics files into a folder on the laptop. At this point I think all I need is a nicer interface to create/edit the vjournal .ics files (though I could just edit them directly I suppose).

**Reply by Lyzod** [reply]
Not to be a dick, but that's the entire problem of this thread. Finding a nice interface for vJournal. vCard and the typical tasks and events are no problem, but the vJournal is. No idea why it isn't that popular, but viewing it proved to be a challenge on top of the inconsistent standards or rather the extensions of the it. Google being a prime example of not following the standards enough that it breaks. Congratz on Degoogling, but you still have a long way to go, especially if you made use of the fields Google is using and not just names and phone numbers. 

jtxboard is not a classical CalDAV client, bc it doesn't show you events, it's focused on the vNote and vJournal part, complimenting a standard calendar app like Fossify Calendar or Etar (if you want to export/import a file from a server on mobile, you need both). 

DAVx5 is the standard (and pretty much competitionless) to pull WebDAV and CalDAV content including that proprietary nonsense the Big 5 are cooking, so you are on the right path here. 

For desktop however, as you can see in this thread, the implementation options all kinda suck bc they are ignoring parts of the data (mainly hierarchy), leaving a mess, or are completely unreliable. It's the point of this thread to find a parser for the file(s) that works well. 

On the topic of syncing, you can consult the jtxboard website, which server reliably supports vJournal and vNote, but spoiler: All current solutions only give you backend support, nothing that looks good and let's you edit. Most people that go that extra mile to degoogle and look for supports for file standards, ics and vCard etc in this case, know enough to open such a file to edit it; opening it in a normal text editor without bells is not a big problem. 

CalDAV and its associated files are only really usable if you don't need to deal with the entirety of the whole record file directly. Even with a search utensil that's just not feasible. 

I need to start learning how the standard works and how to build a useful interface around it. I am thinking that I would probably going to use Emacs as a framework for this. It already has ical support and vCard support and I think writing a syncing solution to the CalDAV Server should be possible, taking inspiration from the existing support and building upon org journal and org diary bc why not, really? Seems analogous. It also has the obvious benefit of perfectly integrating into the environment, so I can refer to contacts, events, bibliography, sending emails, pasting from RSS feeds and so on. I am aware that not everyone will be willing to learn emacs to be able to use it, but yeah, I can't fix the world. The most I could offer is to package it, so you can run the parser into plain text files without needing to run emacs, so you can use the resulting org journal files wherever you want. The viewer will probably build off emacs functions too much to be able to be packaged individually, but maybe someone else is willing to implement this. 

Currently that's just future plans, I gotta plan other stuff including my medical studies first.

**Reply by ergodicbreak** [reply]
> I am thinking that I would probably going to use Emacs as a framework for this.

Maybe it's no coincidence but I've been leaning this way as well after looking at options today. I already use Emacs for most text editing. I've been trialing Thunderbird for mail and calendar just because in the past putting everything (tm) in Emacs has been a bit much. I think Emacs combined with vdirsyncer could work well (and indeed someone wrote a package called cdsync to integrate CalDav with Emacs diary). 

However the big find of the day was that someone has written an icalendar parser recently meant to improve upon the existing parsers in Emacs. You can find discussion on emacs-devel and the repo is at https://git.sr.ht/~wyleyr/icalendar-el. He also wrote a long post on it here https://recursewithless.net/emacs/icalendar-parser-and-mode.org.

Also no worries on tone and I appreciate the info! This landscape is something of a mess it seems and any info is helpful.

**Reply by Lyzod** [reply]
Interesting finds, but if I am reading it correctly, sadly not the end yet, but knowing how easy it is to write custom functions and views in emacs, it's just a matter of a few pages worth of elisp code to add the functionality. Rendering a view will be the hardest, but as said, I want to build off existing implementations. 

If I see that correctly, the cdsync is just an alternative to orgmode - ical sync. The icalendar parser is more interesting as it creates syntax highlighting for the file itself and it does mention vJournal once. As said, I don't think, if you have a massive file, that having just a list of entries is really usable, at least my brain just wants to see parts of it and that parser doesn't make it accessible to org roam at all. 

We need a viewer, also in Calendar view, a sync component, and I want to recreate jtx boards features of toggling vNotes into vJournals and vice versa and associate vNotes with vTasks and events etc. 

I am working on an overengineered emacs based CRM with communication logs, objects with RDF entities that partially syncs with my CalDAV server, so I can access the most important part of the data in any app. Sadly I am just at the modelling stage, so this will take A WHILE.

**Reply by Julianoe** [reply]
Just wanted to point that on Dekstop linux I've been using kOrganizer since I stumbled upon this very thread of discussion and the more I use it, the more I like it. It takes a little bit of getting used to but now I use it fully for all my calendars, following calendars from other people or third places events, editing my journal entries etc.
To be honest I rarely edit from jtxBoard because I find it not comfortable at all for that. Its main quality is pushing for a more broad use of the CalDav standard and vJournal but other than that it's no a UX/UI I want to spend time in writing. A shame.

**Reply by rogercreagh** [reply]
Interesting @Julianoe - I am almost the reverse, At present I only use Kontact to be able to view and partially edit journal entries, but use other apps on linux for calendar, email, tasks etc.

I wonder if you are trying to create/edit entries on a phone - I have a dedicated tablet with a e-ink screen for journals (jtxB) and books (kindle) and drawingi notes (a built in boox app) and find jtxB perfectly good for editing journals on a handheld device - I couple of quirks that could be ironed out (layout of tag/category selection when editing being the biggest bugbear - there was an issue for this but nothing has happened, probably because the focus now seems to be on Tasks, which is a shame as there are plenty of alternative tasks app, but none for journals.

I agree a phone is not ideal for creating/editing entries (or anything else come to that!)

For me, the biggest problem with Kontact/Korganiser for journals is that tags/categories are only shown in the edit screen and you filtering/search on them is buried a few layers away. It may be that there are config setting for this but I haven't found them.

**Reply by Lyzod** [reply]
Guys, respectfully, you are in the side thread. This doesn't relate to it. Go back to the main discussion if at all.

**Comment by baumea** [comment]
If someone is interested, I just wrote a fzf-based terminal application for that purpose: https://github.com/baumea/fzf-vjour

I am extremely appy to use jtxBoard on my phone, but was not satisfied with any client for my computer; hence this effort now. Hope it could be of help to some of you.
